### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Yuqiu-Yang**
The prediction is done by first grouping pmhcs. For each pmhc and each model, one batch of background tcrs will be used so that the logits will be concordant with the ranks

### CODE REVIEW

**Comment by Yuqiu-Yang** [comment]
Merging dev to main
